File: backend/model/Bill.js
const mongoose = require("mongoose");

const BillSchema = new mongoose.Schema({
  roommate_group_id: { type: mongoose.Schema.Types.ObjectId, ref: "RoommateGroup", required: true },
  submitter_id: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  // TODO: Add back in
  // grocery_item_id: { type: mongoose.Schema.Types.ObjectId, ref: "GroceryItem", required: true },
  amount: { type: Number, required: true },
  date_submitted: { type: Date, default: Date.now },
  status: { type: String, enum: ["paid", "unpaid"], default: "unpaid" }
});

module.exports = mongoose.model("Bill", BillSchema);

File: App/Roommate App/Models/Bill.swift
//
//  Bill.swift
//  Roommate App
//
//  Created by Logan Norman on 4/5/23.
//

import Foundation

enum Status {
    case Paid
    case UnPaid
}

class Bill {
    let submitter: User
    let amount: Float
    let date: Date
    let status: Status
    
    init(submitter: User, amount: Float, date: Date, status: Status) {
        self.submitter = submitter
        self.amount = amount
        self.date = date
        self.status = status
    }
}
